mono/metadata/native-library.c

```diff
@@ -1203,14 +1203,15 @@ legacy_probe_for_module (MonoImage *image, const char *new_scope)
 	if (mono_get_find_plugin_callback ())
 	{
 		const char* unity_new_scope = mono_get_find_plugin_callback () (new_scope);
-		if (unity_new_scope == NULL)
+		if (unity_new_scope == NULL || !unity_new_scope[0])
 		{
 			mono_trace (G_LOG_LEVEL_WARNING, MONO_TRACE_DLLIMPORT,
 				"DllImport unable to load unity mapped library '%s'.",
 				unity_new_scope);
 		}
 
-		new_scope = g_strdup (unity_new_scope);
+		else
```

Comment:
Why do we need the else?

Reply:
new_scope needs to remain unchanged if unity_new_scope is NULL or "".